sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/JpaCriteriaPaginatedBoxableSource.java

```diff
@@ -0,0 +1,59 @@
+package uk.ac.bbsrc.tgac.miso.persistence.impl;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.function.Consumer;
+
+import javax.persistence.criteria.Join;
+import javax.persistence.criteria.Path;
+
+import uk.ac.bbsrc.tgac.miso.core.data.AbstractBoxPosition;
+import uk.ac.bbsrc.tgac.miso.core.data.Box;
+import uk.ac.bbsrc.tgac.miso.core.data.Boxable;
+import uk.ac.bbsrc.tgac.miso.core.util.DateType;
+import uk.ac.bbsrc.tgac.miso.core.util.LimsUtils;
+
+public interface JpaCriteriaPaginatedBoxableSource<T extends Boxable> extends JpaCriteriaPaginatedDataSource<T, T> {
+
+  @Override
+  default void restrictPaginationByBox(QueryBuilder<?, T> builder, String query, Consumer<String> errorHandler) {
+    Join<T, ? extends AbstractBoxPosition> join = builder.getSingularJoin(builder.getRoot(), "boxPosition", null);
+    if (LimsUtils.isStringBlankOrNull(query)) {
+      builder.addPredicate(builder.getCriteriaBuilder().isNull(join.get("box")));
+    } else {
+      Join<? extends AbstractBoxPosition, Box> boxJoin = builder.getSingularJoin(join, "box", null);
```

Comment:
Use `AbstractBoxPosition_` metamodel and `getJoin` rather than hard-coded Strings and `getSingularJoin`